bundle-workflow/tests/tests_test_workflow/test_dependency_installer.py

```diff
@@ -0,0 +1,93 @@
+import os
+import shutil
+import unittest
+from unittest.mock import call, patch
+
+from manifests.build_manifest import BuildManifest
+from test_workflow.dependency_installer import DependencyInstaller
+
+
+class DependencyInstallerTests(unittest.TestCase):
+    def setUp(self):
+        self.maxDiff = None
+        self.manifest_filename = os.path.join(
+            os.path.dirname(__file__),
+            "../tests_assemble_workflow/data/opensearch-build-1.1.0.yml",
+        )
+        self.manifest = BuildManifest.from_path(self.manifest_filename)
+
+    @patch("test_workflow.dependency_installer.S3Bucket")
+    def test_install_maven_dependencies(self, mock_s3_bucket):
+        s3_bucket = mock_s3_bucket.return_value
+        self.dependency_installer = DependencyInstaller(self.manifest.build)
+        dependencies = dict(
+            {
+                "opensearch-job-scheduler": "1.1.0.0",
+                "opensearch-anomaly-detection": "1.1.0.0",
+            }
+        )
+        self.dependency_installer.install_maven_dependencies(dependencies)
+        self.assertEqual(s3_bucket.download_folder.call_count, 2)
+        maven_local_paths = []
+        for dependency, version in dependencies.items():
+            maven_local_paths.append(
+                os.path.join(
+                    os.path.expanduser("~"),
+                    f".m2/repository/org/opensearch/{dependency}/{version}/",
+                )
+            )
+        s3_bucket.download_folder.assert_has_calls(
+            [
+                call(
+                    "opensearch-job-scheduler/1.1.0.0",
+                    maven_local_paths[0],
```

Comment:
This will become something like `dependency_installer.maven_local_path('opensearch-job-scheduler')`